test/extensions/filters/http/credential_injector/filter_test.cc

```diff
@@ -0,0 +1,145 @@
+#include "source/extensions/filters/http/credential_injector/credential_injector_filter.h"
+#include "source/extensions/http/injected_credentials/generic/generic_impl.h"
+
+#include "test/mocks/http/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CredentialInjector {
+class MockSecretReader : public Http::InjectedCredentials::Common::SecretReader {
+public:
+  MockSecretReader(const std::string& secret) : secret_(secret){};
+  const std::string& credential() const override { return secret_; }
+
+private:
+  const std::string secret_;
+};
```

Comment:
nit: I think it would be better to mock the `CredentialInjector` rather than to mock the `SecretReader`. Because the filter actually only use the `CredentialInjector` and the `SecretReader` is only part of implementation of `GenericCredentialInjector`.